sisyphus/job.py

```diff
@@ -133,7 +133,7 @@ def __call__(cls: Type[T], *args, **kwargs) -> T:
 
         # add stacktrace information, if set to None or -1 use full stack
         stack_depth = gs.JOB_ADD_STACKTRACE_WITH_DEPTH + 1 if gs.JOB_ADD_STACKTRACE_WITH_DEPTH >= 0 else None
-        if stack_depth > 1 or None:
+        if stack_depth is None or isinstance(stack_depth, int) and stack_depth > 1:
```

Comment:
Why do we need the isinstance check? It seems to me that it just unnecessary narrows down the functionality
```suggestion
        if stack_depth is None or stack_depth > 1:
```

Reply:
Yeah, what you propose makes sense, it was just to "assert" that it was an integer.